dvc/main.py

```diff
@@ -50,8 +74,9 @@ def main(argv=None):  # noqa: C901
 
         logger.trace(args)
 
-        cmd = args.func(args)
-        ret = cmd.run()
+        with profile(enable=args.profile):
+            cmd = args.func(args)
+            ret = cmd.run()
```

Comment:
arguably analytics code down below might also be of some interest to us :smiley: